api/v1alpha1/hostdevicenetwork_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 NVIDIA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	HostDeviceNetworkCRDName = "HostDeviceNetwork"
+)
+
+// HostDeviceNetworkSpec defines the desired state of HostDeviceNetwork
+type HostDeviceNetworkSpec struct {
+	// Namespace of the NetworkAttachmentDefinition custom resource
+	NetworkNamespace string `json:"networkNamespace,omitempty"`
+	//  configuration to be used for this network.
+	IPAM string `json:"ipam,omitempty"`
+}
+
+// HostDeviceNetworkStatus defines the observed state of HostDeviceNetwork
+type HostDeviceNetworkStatus struct {
+	// Reflects the state of the MacvlanNetwork
+	// +kubebuilder:validation:Enum={"notReady", "ready", "error"}
+	State State `json:"state"`
+	// Network attachment definition generated from HostDeviceNetworkSpec
+	HostDeviceNetworkAttachmentDef string `json:"hostDeviceNetworkAttachmentDef,omitempty"`
+	// Informative string in case the observed state is error
+	Reason string `json:"reason,omitempty"`
+	// AppliedStates provide a finer view of the observed state
+	AppliedStates []AppliedState `json:"appliedStates,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// kubebuilder:object:generate
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.state`,priority=0
+// +kubebuilder:printcolumn:name="Age",type=string,JSONPath=`.metadata.creationTimestamp`,priority=0
+
+// MacvlanNetwork is the Schema for the macvlannetworks API
+type HostDeviceNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   *HostDeviceNetworkSpec  `json:"spec,omitempty"`
```

Comment:
why the Spec is with *? 
I don't see it like this in https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/api/v1/sriovibnetwork_types.go#L49

Reply:
Fixes